src/Builder/BuilderEncoder.php

```diff
@@ -101,6 +102,10 @@ public function encode($value): stdClass|array|string
             throw new LogicException(sprintf('Class "%s" does not implement OperatorInterface.', $value::class));
         }
 
+        if ($value instanceof BackedEnum) {
+            return $this->encodeAsEnum($value);
```

Comment:
The naming of this method seems odd to me, as the other `encodeAs` methods are consistent with `$value::ENCODE`. Would `encodeEnum()` make more sense here? Alternatively, I think the method might not even be required at all.

I'm was a bit confused by the difference in how the `Encode::Array` case is handled for a BackedEnum vs. the `switch` below. Looking at `encodeAsArray()`, it creates a list of the object's properties. In `encodeAsEnum()`, you explicitly extract the `value` property (presumably because the `name` property is irrelevant) and then apply the handling we see in `encodeAsArray()` or `encodeAsSingle()`.

Would it make more sense to eagerly convert the BackedEnum into an object with a single `value` property, and then feed that to the switch statement below?

I imagine a BackedEnum has no business being used for some other encoding types, such as `Encode::Group`, but I'm not sure if you'd want to catch or otherwise assert for that up front or trust the existing switch logic to produce something that server would ultimately reject.

Reply:
Thanks for the review. I found a simpler implementation by only checking if `$value` is a `BackedEnum` in `encodeAsArray`. I don't need to implement other encoding methods for now, as it's the only enum I have.